metrics/repl.lag/lag.go

```diff
@@ -198,3 +259,58 @@ func (c *Lag) Collect(ctx context.Context, levelName string) ([]blip.MetricValue
 	}
 	return []blip.MetricValue{m}, nil
 }
+
+func (c *Lag) collectPFS(ctx context.Context, levelName string) ([]blip.MetricValue, error) {
+	var defaultLag []blip.MetricValue
+	if c.dropNotAReplica[levelName] {
+		defaultLag = nil
+	} else {
+		// send -1 for lag
+		m := blip.MetricValue{
+			Name:  "current",
+			Type:  blip.GAUGE,
+			Value: float64(-1),
+		}
+		defaultLag = []blip.MetricValue{m}
+	}
+
+	// if isReplCheck is supplied, check if it's a replica
+	isRepl := 1
+	if c.replCheck != "" {
+		query := "SELECT @@" + c.replCheck
+		if err := c.db.QueryRowContext(ctx, query).Scan(&isRepl); err != nil {
+			return nil, fmt.Errorf("checking if instance is replica failed, please check value of %s. Err: %s", OPT_REPL_CHECK, err.Error())
+		}
+	}
+
+	if isRepl == 0 {
+		return defaultLag, nil
+	}
+	// instance is a replica or replCheck is not set
+
+	var lagValue sql.NullString
+	if err := c.db.QueryRowContext(ctx, MySQL8LagQuery).Scan(&lagValue); err != nil {
+		return nil, fmt.Errorf("could not check replication lag, check that this is a MySQL 8.0 replica, and that performance_schema is enabled. Err: %s", err.Error())
+	}
+	if !lagValue.Valid {
+		// it is a MySQL 8 instance and performance schema is enabled, otherwise the query would have returned error
+		// if replCheck is empty, we can assume based on the query that it's not a replica and return nil or -1
+		if c.replCheck == "" {
+			return defaultLag, nil
+		} else {
+			// it's a replica, so lagValue should be valid, raise error
+			return nil, fmt.Errorf("instance is a MySQL 8 replica and performance schema is enabled, still could not calculate lag, please check manually. Lag: %q", lagValue.String)
```

Comment:
To make error message more clear (because user can't/won't check manually): "cannot determine replica lag because performance_schema.replication_applier_status_by_worker returned an invalid value: %v (expected a positive integer value)"